File: test/react-native/features/fixtures/ios-module/Scenarios/MetadataNativeUnhandledScenario.m
#import "MetadataNativeUnhandledScenario.h"

@implementation MetadataNativeUnhandledScenario

- (void)run: (RCTPromiseResolveBlock)resolve
     reject:(RCTPromiseRejectBlock)reject {
  NSException *exception = [[NSException alloc] initWithName:@"NSException" reason:@"MetadataNativeUnhandledScenario" userInfo:nil];
  [Bugsnag addMetadata:@"set via client" withKey:@"some_more_data" toSection:@"nativedata"];
  @throw exception;

}

@end

File: test/react-native/features/fixtures/reactnative/scenarios/MetadataNativeUnhandledScenario.kt
package com.reactnative.scenarios

import android.content.Context
import com.bugsnag.android.Bugsnag
import com.bugsnag.android.OnErrorCallback

class MetadataNativeUnhandledScenario(context: Context): Scenario(context) {

    override fun run() {
        super.run()
        Bugsnag.addMetadata("nativedata", "some_more_data", "set via client")
        Bugsnag.addOnError(OnErrorCallback { event ->
            event.addMetadata("nativedata", "even_more_data", "set via event")
            true
        })
        throw generateException()
    }
}
